tsl/test/isolation/specs/remote_create_chunk.spec.in

```diff
@@ -0,0 +1,127 @@
+# Please see the included NOTICE for copyright information and
+# LICENSE-TIMESCALE for a copy of the license.
+
+# Test concurrent remote chunk creation
+#
+
+setup
+{
+	CREATE TABLE conditions (time timestamptz, device int, temp float);
+	SELECT create_hypertable('conditions', 'time', 'device', 2, chunk_time_interval => interval '1 day');
+
+	CREATE OR REPLACE FUNCTION debug_waitpoint_enable(TEXT) RETURNS VOID LANGUAGE C VOLATILE STRICT
+	AS '@TS_MODULE_PATHNAME@', 'ts_debug_waitpoint_enable';
+
+	CREATE OR REPLACE FUNCTION debug_waitpoint_release(TEXT) RETURNS VOID LANGUAGE C VOLATILE STRICT
+	AS '@TS_MODULE_PATHNAME@', 'ts_debug_waitpoint_release';
+
+	CREATE OR REPLACE FUNCTION debug_waitpoint_id(TEXT) RETURNS BIGINT LANGUAGE C VOLATILE STRICT
+	AS '@TS_MODULE_PATHNAME@', 'ts_debug_waitpoint_id';
+
+	CREATE OR REPLACE FUNCTION waitpoint_count(tag TEXT) RETURNS bigint AS
+	$$
+	SELECT count(*)
+	FROM pg_locks
+	WHERE locktype = 'advisory'
+	AND mode = 'ShareLock'
+	AND objid = debug_waitpoint_id(tag)
+	AND granted = false;
+	$$ LANGUAGE SQL;
+
+	CREATE OR REPLACE FUNCTION root_table_lock_count() RETURNS bigint AS
+	$$
+	SELECT count(*) AS num_locks_on_conditions_table
+	FROM pg_locks
+	WHERE locktype = 'relation'
+	AND relation = 'conditions'::regclass
+	AND mode = 'ShareUpdateExclusiveLock';
+	$$ LANGUAGE SQL;
+}
+
+teardown {
+	DROP TABLE conditions;
+}
+
+
+session "s1"
+setup	{
+	SET TRANSACTION ISOLATION LEVEL READ COMMITTED;
+	SET LOCAL lock_timeout = '500ms';
+	SET LOCAL deadlock_timeout = '300ms';
+	SET application_name = 's1';
+}
+
+# Try to create a chunk
+step "s1_create_chunk_1" {
+	SELECT slices, created FROM _timescaledb_internal.create_chunk('conditions', jsonb_build_object('time', ARRAY[1514764800000000, 1514851200000000], 'device', ARRAY[-9223372036854775808, 1073741823]));
+}
+
+step "s1_create_chunk_2" {
+	SELECT slices, created FROM _timescaledb_internal.create_chunk('conditions', jsonb_build_object('time', ARRAY[1514764800000000, 1514851200000000], 'device', ARRAY[1073741823, 9223372036854775807]));
+}
+
+# Create a chunk that does not exist
+session "s2"
+setup	{
+	SET TRANSACTION ISOLATION LEVEL READ COMMITTED;
+	SET LOCAL lock_timeout = '500ms';
+	SET LOCAL deadlock_timeout = '300ms';
+	SET application_name = 's1';
+}
+
+step "s2_create_chunk_1" {
+	SELECT slices, created FROM _timescaledb_internal.create_chunk('conditions', jsonb_build_object('time', ARRAY[1514764800000000, 1514851200000000], 'device', ARRAY[-9223372036854775808, 1073741823]));
+}
+
+step "s2_create_chunk_2" {
+	SELECT slices, created FROM _timescaledb_internal.create_chunk('conditions', jsonb_build_object('time', ARRAY[1514764800000000, 1514851200000000], 'device', ARRAY[1073741823, 9223372036854775807]));
+}
+
+session "s3"
+setup {
+	SET application_name = 's3';
+}
+
+step "s3_conditions_locks" {
+
+	SELECT root_table_lock_count() AS table_lock,
+	waitpoint_count('find_or_create_chunk_started') AS at_start,
```

Comment:
This tag doesn't match the enabled tags. Should it be:
```suggestion
	waitpoint_count('find_or_create_chunk_start') AS at_start,
```

Reply:
Good catch! Will fix.